src/@next/components/organisms/CartRow/CartRow.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useState } from "react";
+
+import { Icon, IconButton } from "@components/atoms";
+import { CachedImage, TextField } from "@components/molecules";
+
+import * as S from "./styles";
+import { IProps } from "./types";
+
+const QuantityButtons = (
+  add: () => void,
+  substract: () => void,
+  index?: number
+) => (
+  <S.QuantityButtons>
+    <div
+      onClick={substract}
+      data-cy={`cartPageItem${index}QuantityBtnSubtract`}
+    >
+      <Icon size={16} name="horizontal_line" />
+    </div>
+    <div onClick={add} data-cy={`cartPageItem${index}QuantityBtnAdd`}>
+      <Icon size={16} name="plus" />
+    </div>
+  </S.QuantityButtons>
+);
+
+/**
+ * Product row displayed on cart page
+ */
+export const CartRow: React.FC<IProps> = ({
+  index,
+  totalPrice,
+  unitPrice,
+  name,
+  sku,
+  quantity,
+  maxQuantity,
+  onQuantityChange,
+  thumbnail,
+  attributes = [],
+  onRemove,
+}: IProps) => {
+  const [tempQuantity, setTempQuantity] = useState<number | string>(quantity);
+  const [isTooMuch, setIsTooMuch] = useState(false);
+
+  const handleBlurQuantityInput = () => {
+    const newQuantity =
+      typeof tempQuantity === "number"
+        ? tempQuantity
+        : parseInt(tempQuantity, 10);
+    const notEnoughQuantity = isNaN(newQuantity) || newQuantity <= 0;
+    const tooMuchQuantity = newQuantity > maxQuantity;
+    if (notEnoughQuantity && !tooMuchQuantity) {
+      setTempQuantity(quantity);
```

Comment:
No onQuantityChange(maxQuantity); needed?

Reply:
No, it simply reset `tempQuantity` to the one before changes - the `quantity` from props.